### PR TITLE
Scanlndhook

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/jhxqum7z0qj4e46d976hccy0rm

[download apk](http://filestorage.bluewallet.io:1488/1607147139.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: screen/lnd/scanLndInvoice.js [hunk 1]
```diff
@@ -29,131 +28,70 @@ import { Icon } from 'react-native-elements';
...
+const ScanLndInvoice = () => {
+  const { wallets, fetchAndSaveWalletTransactions } = useContext(BlueStorageContext);
+  const { colors } = useTheme();
+  const { walletID, uri, invoice } = useRoute().params;
+  const name = useRoute().name;
+  /** @type {LightningCustodianWallet} */
+  const wallet = useRef(
+    wallets.find(item => item.getID() === walletID) || wallets.find(item => item.type === LightningCustodianWallet.type),
+  );
+  const { navigate, dangerouslyGetParent, setParams, pop } = useNavigation();
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/lnd/scanLndInvoice.js:44]: Its there. Line 73.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/y7wqceguhbj75233rbpj2h1x4m

[download apk](http://filestorage.bluewallet.io:1488/1607359194.apk)

**Review by limpbrains — APPROVED** [review]

File: screen/lnd/scanLndInvoice.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply screen/lnd/scanLndInvoice.js:44]: ok I see code that handles errof if there is no ln wallets.
but where is the code that chooses first found ln wallet if walletID not passed or we could not find anything by walletID?

[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/lnd/scanLndInvoice.js:44]: Line 42.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n5h7fur47qnd5vzbnpkp616674

[download apk](http://filestorage.bluewallet.io:1488/1607379638.apk)

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/reqmbd189bkf91qmpae1tc0a48

[download apk](http://filestorage.bluewallet.io:1488/1607453570.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dp7gcz6nmecq453thpfa6tj45r

[download apk](http://filestorage.bluewallet.io:1488/1607472235.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8x1tb4arf3fjhwt1d0ke9f2wpr

[download apk](http://filestorage.bluewallet.io:1488/1607659764.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gjfw0499jr59c6yt4354vgzpp0

[download apk](http://filestorage.bluewallet.io:1488/1607730238.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/54dpqf6qy1hatz0frr9dpdc028

[download apk](http://filestorage.bluewallet.io:1488/1607736996.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/t4z2kh0bfhxnj1nbmau12286em

[download apk](http://filestorage.bluewallet.io:1488/1607755633.apk)

**Pull request merged**